### PR TITLE
library: Ported `LinkButton` in Rust

### PR DESCRIPTION
**Pull request by mbilal234**
Please check the rust implementation of the linkbutton.
The preview works fine on the workbench on my side.

Please review.

### CODE REVIEW

File: src/Library/demos/Link Button/code.rs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+use crate::workbench;
+use gtk::prelude::*;
+
+pub fn main() {
+    let linkbutton: gtk::LinkButton = workbench::builder().object("linkbutton").unwrap();
+    linkbutton.connect_clicked(move |_| {
```
Comment (Hofer-Julian) [inline src/Library/demos/Link Button/code.rs:6]: The javascript version uses https://gtk-rs.org/gtk4-rs/stable/latest/docs/gtk4/struct.Button.html#method.connect_notify with the parameter "visited" here.

File: src/Library/demos/Link Button/code.rs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+        println!("Link has been visited");
+    });
+
+    linkbutton.connect_activate_link(|button| {
+        println!("About to activate {}", button.uri());
+
```
Comment (Hofer-Julian) [inline src/Library/demos/Link Button/code.rs:12]: Please add the same comment of the javascript version: https://github.com/sonnyp/Workbench/blob/4169b5d9b27b6bd77bbc10bded7ff77f7fdadb1a/src/Library/demos/Link%20Button/main.js#L10

**Review by Hofer-Julian — COMMENTED** [review]
After the comments are addressed, it looks good to me to merge

File: src/Library/demos/Link Button/code.rs (hunk 2 above)
Comment (mbilal234) [reply src/Library/demos/Link Button/code.rs]: Ive added the comments

File: src/Library/demos/Link Button/code.rs (hunk 1 above)
Comment (mbilal234) [reply src/Library/demos/Link Button/code.rs:6]: I tried with connect_notify first but could not resolve errors, so used an alternative.
It says "closure takes 2 arguments but is given 1" even though i give it 2 arguments.
Ill try using connect_notify and resolve this

Comment (mbilal234) [reply src/Library/demos/Link Button/code.rs:8]: Hello, I've added the method you specified.

Please review :)

File: src/Library/demos/Link Button/code.rs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    linkbutton.connect_clicked(move |_| {
+        println!("Link has been visited");
+    });
+
+    linkbutton.connect_notify(Some("visited"), |button, _| {
+        println!("About to activate {}", button.uri());
+      // Return true if handling the link manually, or
+        // false to let the default behavior continue
+        false;
```
Comment (Hofer-Julian) [inline src/Library/demos/Link Button/code.rs:16]:
```suggestion
```

Seems like the Rust version doesn't allow a return parameter.
So we can just remove this section.

**Review by Hofer-Julian — CHANGES_REQUESTED** [review]

**Comment by Hofer-Julian** [comment]
@mbilal234 could you please remove the separator demo from this PR?

**Comment by mbilal234** [comment]
> @mbilal234 could you please remove the separator demo from this PR?

Deleted

**Review by Hofer-Julian — APPROVED** [review]

**Pull request merged**